src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -28,11 +28,13 @@ public int basketCost() {
     public void contentsOfTheBasket() {
         boolean basketIsEmpty = true;
         int countOfSpecialProducts = 0;
-        for (Product product : products) {
-            if (product != null) {
-                System.out.println(product);
-                countOfSpecialProducts = product.isSpecial() ? countOfSpecialProducts + 1 : countOfSpecialProducts;
-                basketIsEmpty = false;
+        for (ArrayList<Product> productList : products.values()) {
```

Comment:
```suggestion
        for (List<Product> productList : products.values()) {
```
не забываем про полиморфизм